x-pack/plugins/ingest_manager/server/services/license.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Observable, Subscription } from 'rxjs';
+import { ILicense } from '../../../licensing/server';
+
+class LicenseService {
```

Comment:
I think this whole thing could be replaced by:

```ts
const licenseSubject$ = new BehaviorSubject();
license$.subscribe(licenseSubject$);
licenseSubject$.value // this gives you the current license!
```

That said, this doesn't appear used anywhere? Not sure what your plans are for this, but in route handlers you can already access `context.licensing.license` if you depend on the licensing plugin.

Reply:
This isn't being used for anything currently, it's boilerplate that I thought may be needed for future work. I'd like to keep them accessible outside of route context though. (This plugin will eventually expose a number of interfaces for other plugins to use, so our route handlers are generally light wrappers around various services.)

@jfsiii I would be comfortable using BehaviorSubject for license & [config](https://github.com/elastic/kibana/pull/56567#discussion_r377360549) service and moving them inside `AppContextService` instead of their own services, what do you think?

Reply:
@jen-huang SGTM. I took a shot at it in https://github.com/elastic/kibana/pull/56567/commits/80077aea225cc8cfe0a99985d93abe2e206629c2 & https://github.com/elastic/kibana/pull/56567/commits/71220c9bbc43b70418d63f20cdf164a8bf608a4f

I did a quick check by logging the values of `context.licensing.license` & `appContextService.getLicense()` in a handler and they were identical

Reply:
@jen-huang If we're not using these concepts now, perhaps we delete them and add when they're needed? 

I noticed the `AgentConfigService` methods accept `soClient` as an argument (the EPM equivalents in the other repo do as well) so there's precedent for that style https://github.com/elastic/kibana/blob/b59c81c37ce2e8793a9f9b5054d0daa60a1ce635/x-pack/plugins/ingest_manager/server/services/agent_config.ts#L124-L127